src/client/settings.jsx

```diff
@@ -34,10 +34,12 @@ const saveTimes = [
   { value: 30, label: '30 seconds' },
 ];
 
-export const Settings = ({ show, onClose, setTheme }) => {
+export const Settings = ({ show, onClose, setTheme, editor }) => {
   const handleChange = useCallback((selectedOption) => {
     setTheme(selectedOption.value);
-
+    editor.dispatch({
```

Comment:
Putting this here doesn't feel right. Ideally the state change and its side effects should be completely separate. Ideally the `Settings` component should now need to know anything about the `editor`.

I would suggest to put the `editor.dispatch` call into a `useEffect` that depenes on the `theme` state, in a higher level of the component tree.

To implement the theme changing correctly, we may want to consider invoking this on all editors in the editor cache, not just the open editor.